webdriver-spec.html

```diff
@@ -375,20 +404,79 @@ <h4 id='returning-http-response'>Returning a HTTP response to the local end</h4>
-        the underlying collection of capabilities that a User Agent handle, represented as a dictionary
-        mapping strings to values that can be serialised
-              The default value is an empty dictionary.
-    </li>
-  </ul>
+
+<h2>Sessions</h2>
+
+<p>Non-normative summary:
+ A session is equivalent to a single instantiation of a particular user agent,
+ including all its child browsers.
+ WebDriver gives each session a unique <a>session ID</a>
+ that can be used to differentiate one session from another,
+ allowing multiple user agents to be controlled on the same machine if needed,
+ and allowing sessions to be routed via a multiplexer.
+
+<p>A WebDriver <dfn>session</dfn> represents
+ the connection between a local end and a specific remote end.
+ A remote end that is not an <a>intermediary node</a>
+ must have at most one <a>active session</a> at a given time.
+<!-- Is this actually a requirement? -->
+
+<!--
+ TODO(jgraham): Need to explain that there must be
+ a relationship between teh current session
+ and the browser state.
+-->
+
+<p>The <a>session</a> is set up at the invocation of
+ a <a href=#newsession>new session</a>,
+ and torn down at some later point;
+ either explicitly by invoking <a href=#>quit</a>,
+ or implicitly when <a href=#close>close</a> is called at the last remaining
+ <a href=http://www.w3.org/TR/html5/browsers.html#top-level-browsing-context>top level browsing context</a>.
+
+<p>A <a>session</a> has an associated <dfn>session ID</dfn> (a <a>UUID</a>)
+ used to uniquely identify thi session.
+ Unless stated otherwise it is null.
+
+<p>A <a>session</a> has an associated <dfn>alive flag</dfn>
+ that signifies that the session has not yet been torn down.
+ Unless stated otherwise it is unset.
+
+<p>A <a>session</a> has an associated <dfn>session capabilities</dfn>
+ object (a <a>capabilities</a> object)
+ describing the functionality provided by this <a>session</a>.
+ Unless stated otherwise it is null.
+
+<p>A <a>session</a> has an associated <dfn>current browsing context</dfn>
+ (a reference to <a href=http://www.w3.org/TR/html5/browsers.html#top-level-browsing-context>top level browsing context</a>) that indicates what <a>browsing context</a>
+ commands are executed against.
+
+<p>A <a>session</a> has an associated <dfn>script timeout</dfn> (an integer)
+ that specifies a time in milliseconds to wait for scripts to run.
+
+<p>A <a>session</a> has an associated <dfn>page loading timeout</dfn> (an integer)
+ that specifies a time in milliseconds to wait for the page loading to complete.
+
+<p>A <a>session</a> has an associated <dfn>known sessions</dfn>
+ (a sequence of <a>UUID</a>'s) that list
+ <a rel="active session">sessions that are active in a <a>remote end</a>.
+
+<p>A <a>current session</a> (a <a>UUID</a>)
+ is the <a rel="active session">session that is active</a>
+ for the duration of a <a>command</a>.
+<!-- TODO: Doesn't define what "active" means -->
+
+<p class=note>The concept of sessions
```

Comment:
Actually this isn't quite true (I know I wrote it). The concept of a session allows global state in the remote end that lasts for the duration of a session. The ability to have multiple sessions in a specific remote end allows multiplexing intermediaries. On a related note, defining the behaviour of |new session| probably requires that we allow an implementation-specific |maximum number of sessions|, which we can check against the length of |known sessions| to see if we can start a session or not.